Remove validation exceptions Slider min/max values

jfversluis opened the pull request:
### Description of Change

This PR removes the validation of the `Slider.Minimum` and `Slider.Maximum`. Effectively this means we remove the requirement to specify `Minimum` and `Maximum` in a certain order in XAML. Users are now responsible for setting the right values. If that doesn't happen (e.g. `Minimum` is bigger than `Maximum`) weird behavior might occur (which will be the platforms behavior), but the app will not crash.

I left the exception for the constructor in place as this is still some kind of safeguard and doesn't stand in the way of what we're trying to fix here.

### Issues Fixed

Fixes #6305

jfversluis committed:
Remove validation exceptions Slider min/max values

jsuarezruiz commented on src/Controls/src/Core/Slider.cs:
```diff
@@ -9,31 +8,23 @@ namespace Microsoft.Maui.Controls
-			var slider = (Slider)bindable;
-			return (double)value < slider.Maximum;
-		}, coerceValue: (bindable, value) =>
+		public static readonly BindableProperty MinimumProperty = BindableProperty.Create(nameof(Minimum), typeof(double), typeof(Slider), 0d, coerceValue: (bindable, value) =>
```
If we remove the validations here, sould we include in the handlers a check for nonsensical states when any of the properties are updated? (avoid errors,  settings zeros etc)

jfversluis replied:
I think the only scenario that gets you in trouble right now is when you set the `Minimum` to more than the `Maximum`. From what I tested the thumb will just get stuck. And that is just the default behavior then for that platform. So if I was a iOS or Android developer and I were to do that, I would end up with the same thing. Should we then get in the middle and do something else?

The problem is then, again, the order of things. If we are going to make checks like: minimum is more than maximum and them we make minimum equal to maximum and your setting one before the other, you again end up with behavior you might not want and that is dependant on the order of declaration.

hartez replied:
Can we just have the mapping method set the platform min/max/value to zero if the values aren't valid? Then all the platforms do something consistent when the values don't make sense, and the order doesn't matter.

Redth replied:
I'm inclined not to do this, as the way the handlers are structured, if we want to update everything to 0 when one change is invalid, we may cause the UI to 'skip/jump' by changing the values temporarily until the next property change happens and brings in a set of 'valid' values again.

I think what this branch has now is probably reasonable enough.  It doesn't seem to crash any of the platforms so far (still have iOS to validate).

jfversluis commented on src/Controls/tests/Core.UnitTests/SliderUnitTests.cs:
```diff
@@ -59,14 +59,14 @@ public void TestMaxValueClamp()
 		{
 			var slider = new Slider();
-			Assert.Throws<ArgumentException>(() => slider.Maximum = slider.Minimum - 1);
+			Assert.DoesNotThrow(() => slider.Maximum = slider.Minimum - 1);
```
Yeah I guess I could add some more tests also for negatives and such

Redth committed:
Add scrollview to sample page

Redth committed:
Fix incorrect value used to set max track color

This was using the minimum value which could be null to try and set the max track tint color.  This was causing a crash.

jsuarezruiz commented on src/Core/src/Platform/iOS/SliderExtensions.cs:
```diff
@@ -33,7 +33,7 @@ public static void UpdateMaximumTrackColor(this UISlider uiSlider, ISlider slide
 			{
 				if (uiSlider.TraitCollection.UserInterfaceIdiom != UIUserInterfaceIdiom.Mac)
-					uiSlider.MaximumTrackTintColor = slider.MinimumTrackColor.ToPlatform();
+					uiSlider.MaximumTrackTintColor = slider.MaximumTrackColor.ToPlatform();
```
Nice!
